### DISCUSSION TITLE
How do opcode decoding speedup tables work?

### DISCUSSION THREAD

**Discussion by lhsazevedo** [vhelin/wla-dx#397] [Q&A] [answered]
I having trouble to understand the meaning of the values stored in `g_opcode_n` and `g_opcode_p`.

**Comment by vhelin** [comment, marked as answer]
If I remember correctly:

* g_opcode_n - the number of entries we have in g_opcodes_table starting with this ASCII characher
* g_opcode_p - the index in g_opcodes_table where the entries with this starting character start

A quick optimization I did back around 20+ years ago, we use the first character in the source code and jump using that to the entries in g_opcodes_table that start with that character. But if we have 100 entries that start with character A (like AND...) we'll go through them all and stop as soon as we find a match in the mostly sorted structure...
